x/oracle/keeper/hooks.go

```diff
@@ -0,0 +1,51 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	leveragetypes "github.com/umee-network/umee/x/leverage/types"
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+// Hooks defines a structure around the x/oracle Keeper that implements various
+// Hooks interface defined by other modules such as x/leverage.
+type Hooks struct {
+	k Keeper
+}
+
+var _ leveragetypes.Hooks = Hooks{}
+
+// Hooks returns a new Hooks instance that wraps the x/oracle keeper.
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k}
+}
+
+// AfterTokenRegistered implements the x/leverage Hooks interface. Specifically,
+// it checks if the provided Token should be added to the existing accepted list
+// of assets for the x/oracle module.
+func (h Hooks) AfterTokenRegistered(ctx sdk.Context, token leveragetypes.Token) {
+	acceptList := h.k.AcceptList(ctx)
+
+	var tokenExists bool
+	for _, t := range acceptList {
+		if t.BaseDenom == token.BaseDenom {
+			tokenExists = true
+			break
+		}
+	}
+
+	if !tokenExists {
+		acceptList = append(acceptList, types.Denom{
+			BaseDenom:   token.BaseDenom,
+			SymbolDenom: token.SymbolDenom,
+			Exponent:    token.Exponent,
+		})
+	}
+
+	h.k.SetAcceptList(ctx, acceptList)
```

Comment:
Need to also update bank denom metadata here, since we're not updating the accept list at ballot counting : 
```suggestion
	h.k.SetAcceptList(ctx, acceptList)
	h.k.ApplyAcceptList(ctx, acceptList)
```
We'll have to remove the `updateRequired` logic from `ApplyAcceptList`, and maybe rename ApplyAcceptList -> SetAcceptListMetadata

Reply:
So I took a look at `ApplyAcceptList` again and I'm still not clear on what it does or why we even need it?

I'm convinced that we can just remove `ApplyAcceptList` all together. 

- We already update bank metadata for incoming IBC assets ✅ 
- We already have bank metadata for native Umee assets (at genesis) ✅ 
- We _don't need_ bank metadata for ETH ERC20 native assets ✅

That should cover all our bases. Is there a reason you think we need `ApplyAcceptList`?

Reply:
This was taken from terra's code, I wasn't aware we're already updating bank metadata in other places - we can remove it altogether 👍